### ISSUE TITLE
Weight dimensions not documented or checked in degridder

### ISSUE THREAD

**Issue by jfunction** [ska-sa/codex-africanus#161]
* Codex Africanus version:
0.2.0
* Python version:
3.6.8 (default, Aug  7 2019, 17:28:10) \n[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
* Operating System:
Centos7 
Linux 3.10.0-957.21.3.el7.x86_64.debug
### Description

**Describe what you were trying to get done.**
I was trying to test the degridding of some basic data

**Tell us what happened, what went wrong, and what you expected to happen.**
Array values are inconsistent across program runs but consistent within the same program run. This indicates reference to what should be out of bounds memory. 

### What I Did

```python3
import africanus.gridding.simple
import africanus.filters
import numpy as np

nrow = 29
nchan = 1
ncor = 1

sinc = africanus.filters.conv_filters.convolution_filter(3, 7, 'sinc')
grid = np.array(np.sin(np.linspace(0, 10 * 2 * np.pi, 100)), dtype=np.complex64).reshape((10, 10, ncor))

uvw = nrow*[[0, 0, 0], ]  # You can try more reasonable values here, it still breaks
uvw = np.array(uvw, dtype=np.float64)

weights = np.array([np.ones_like(uvw, dtype=np.float64), ])
ref_wave = np.ones(nchan, dtype=np.float64)
cell_size = 1.0

# Degridding is constant in same program run, but varies between program runs:
degridded = africanus.gridding.simple.degrid(grid, uvw, weights, ref_wave, sinc, cell_size)
print(degridded[0, :, 0])
# If we do the same thing within this program run we get the same results:
degridded = africanus.gridding.simple.degrid(grid, uvw, weights, ref_wave, sinc, cell_size)
print(degridded[0, :, 0])
# Rerun the program and get different outputs. 
```

**Comment by sjperkins** [comment]
Thanks for the report @jfunction.

Weights should have a (row, chan, corr) schema. Try setting weights as

```
weights = np.ones((uvw.shape[0], nchan, ncor), dtype=np.float64)
```

I think this is missing in the docs and the code is missing some dimensionality checks.

Note that this is not a very accurate gridder/degridder. May I ask what you're intending to use it for?

**Comment by jfunction** [comment]
Ah yes that's fixed it thanks! 

I'm guessing you want those checks around here? https://github.com/ska-sa/codex-africanus/blob/master/africanus/gridding/simple/gridding.py#L289

And docs for weights changed here? https://github.com/ska-sa/codex-africanus/blob/master/africanus/gridding/simple/gridding.py#L271

I thought this would be an easy to use reference implementation to verify correctness of my degridder. It installed without any issues so I figured that getting it onto other machines shouldn't be a problem. Would you recommend DDFacet instead?

**Comment by sjperkins** [comment]
I've put in PR https://github.com/ska-sa/codex-africanus/pull/162 to force the checks.

However, I might suggest using the nifty gridder/degridder available at https://gitlab.mpcdf.mpg.de/ift/nifty_gridder. It's a very accurate and fast w-stacking implementation.

We have [dask wrappers](https://codex-africanus.readthedocs.io/en/latest/gridding-api.html#nifty) for it in codex, but that may be overkill for your purposes.

**Comment by jfunction** [comment]
:pray: wonderful, thank you! I got a little nervous when I read "Highly experimental pre-release version!" but I'll give it a go. Nothing wrong with using Dask locally without exploiting all of its functionality I guess? I like the API codex africanus provides. 

Thanks again. I guess you'll want to close the issue now?

**Comment by sjperkins** [comment]
>  Nothing wrong with using Dask locally without exploiting all of its functionality I guess? I like the API codex africanus provides.

In that case you may want to combine the codex nifty wrappers with [dask-ms](https://dask-ms.readthedocs.io/en/latest/).

**Comment by sjperkins** [comment]
Closed by #162
